### PR TITLE
ROX-16743: Refactor Namespace Properties to interface

### PR DESCRIPTION
**Pull request by keyallis**
## Description

Refactor to enable notifiers which resolve namespace properties to have different implementations for doing so based on whether they're running in central, sensor, or admission controller.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added

### CODE REVIEW

File: central/notifiers/namespace_properties.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package notifiers
```
Comment (clickboo) [inline central/notifiers/namespace_properties.go:1]: Assuming you will move this to pkg in the second phase of reactor yes?

File: central/notifiers/slack/slack.go [hunk 2]
```diff
@@ -37,8 +37,8 @@ type slack struct {
 	*storage.Notifier
 	client *http.Client
 
-	namespaces namespaceDataStore.DataStore
-	mitreStore mitreDataStore.MitreAttackReadOnlyDataStore
+	namespaceProperties notifiers.NamespaceProperties
+	mitreStore          mitreDataStore.MitreAttackReadOnlyDataStore
```
Comment (clickboo) [inline central/notifiers/slack/slack.go:41]: Will need changes after rebase (mitre store name)

File: central/notifiers/namespace_properties.go (hunk 1 above)
Comment (keyallis) [reply central/notifiers/namespace_properties.go:1]: Yes

File: central/notifiers/slack/slack.go (hunk 2 above)
Comment (keyallis) [reply central/notifiers/slack/slack.go:41]: Can do, I'll wait for your changes and then rebase

**Review by clickboo — APPROVED** [review]
lgtm, thanks for being patient with the muttiple renames!

**Pull request merged**